File: app/src/main/java/me/ycdev/android/devtools/sampler/mem/AppMemStat.java
package me.ycdev.android.devtools.sampler.mem;

import android.util.SparseArray;

public class AppMemStat {
    public String pkgName;
    public SparseArray<ProcMemStat> procSetStats = new SparseArray<ProcMemStat>();

    public AppMemStat(String pkgName) {
        this.pkgName = pkgName;
    }

    public int getTotalPss() {
        int totalPss = 0;
        final int N = procSetStats.size();
        for (int i = 0; i < N; i++) {
            ProcMemStat pidStat = procSetStats.valueAt(i);
            totalPss += pidStat.memPss;
        }
        return totalPss;
    }

    public int getTotalPrivate() {
        int totalRss = 0;
        final int N = procSetStats.size();
        for (int i = 0; i < N; i++) {
            ProcMemStat pidStat = procSetStats.valueAt(i);
            totalRss += pidStat.memPrivate;
        }
        return totalRss;
    }
}

File: app/src/main/java/me/ycdev/android/devtools/sampler/mem/AppMemStat.kt
package me.ycdev.android.devtools.sampler.mem

import android.util.SparseArray

class AppMemStat(var pkgName: String) {
    var procSetStats = SparseArray<ProcMemStat?>()

    val totalPss: Int
        get() {
            var totalPss = 0
            for (i in 0 until procSetStats.size()) {
                val pidStat = procSetStats.valueAt(i)
                totalPss += pidStat!!.memPss
            }
            return totalPss
        }

    val totalPrivate: Int
        get() {
            var totalRss = 0
            for (i in 0 until procSetStats.size()) {
                val pidStat = procSetStats.valueAt(i)
                totalRss += pidStat!!.memPrivate
            }
            return totalRss
        }
}
